CMakeLists.txt

```diff
@@ -182,7 +184,10 @@ IF(CPUINFO_SUPPORTED_PLATFORM)
       ELSEIF(CMAKE_SYSTEM_PROCESSOR MATCHES "^(aarch64|arm64)$")
         LIST(APPEND CPUINFO_SRCS src/arm/linux/aarch64-isa.c)
       ENDIF()
-    ELSEIF(IOS OR (CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND CPUINFO_TARGET_PROCESSOR STREQUAL "arm64"))
+    ELSEIF(IOS OR
```

Comment:
would it be possible to rename `IOS` to `APPLE_MOBILE_OS` and then set `APPLE_OS` = (iOS or tvOS or watchOS), and avoid having to check the specific strings in multiple places?

Reply:
Regular expression matching came into my mind. Does it make sense?

Reply:
i was mostly trying to find a way to put the (ios or watchos or tvos) logic in one place and store that into a variable and then everywhere else just check the variable.

That's unless, of course, you wanted different behavior for the different os's somewhere...

Reply:
Thank you for the clarification @wconstab . Done. Please take another look. Thanks!

Reply:
looks better to me! thanks.

Let me know if CI has issues that aren't your fault- gh actions has been unstable recently.  If its passing CI I think its good to land.  cc @malfet